### PR TITLE
Integrate hist method of Random Forest

### PR DESCRIPTION
**Pull request by agorshk**

### CODE REVIEW

File: daal4py/sklearn/ensemble/_forest.py [hunk 1]
```diff
@@ -481,7 +499,9 @@ def __init__(self,
                  warm_start=False,
                  class_weight=None,
                  ccp_alpha=0.0,
-                 max_samples=None):
+                 max_samples=None,
+                 maxBins=256,
+                 minBinSize=5):
```
Comment (SmirnovEgorRu) [inline daal4py/sklearn/ensemble/_forest.py:504]: probably, better to set minBinSize to 1 by default to work good on small data sets

File: daal4py/sklearn/ensemble/_forest.py [hunk 2]
```diff
@@ -326,7 +342,7 @@ def _daal_fit_regressor(self, X, y, sample_weight=None):
     # create algorithm
     dfr_algorithm = daal4py.decision_forest_regression_training(
         fptype=getFPType(X),
-        method='defaultDense',
+        method='hist',
```
Comment (PetrovKP) [inline daal4py/sklearn/ensemble/_forest.py:345]: there is no hist in the old daal. We must support the previous version (checked in public CI)

**Review by PetrovKP — COMMENTED** [review]
Need to add examples for daal4py with hist method.

File: daal4py/sklearn/ensemble/_forest.py (hunk 1 above)
Comment (agorshk) [reply daal4py/sklearn/ensemble/_forest.py]: it was fixed

File: daal4py/sklearn/ensemble/_forest.py (hunk 2 above)
Comment (agorshk) [reply daal4py/sklearn/ensemble/_forest.py]: correct version for 'hist' was added, examples were added

**Review by SmirnovEgorRu — APPROVED** [review]

**Review by OnlyDeniko — APPROVED** [review]

File: tests/test_examples.py [hunk 3]
```diff
@@ -177,10 +177,14 @@ def test_svm_batch(self):
     ('cholesky_batch', 'cholesky_batch.csv', 'choleskyFactor'),
     ('covariance_batch', 'covariance.csv', 'covariance'),
     ('covariance_streaming', 'covariance.csv', 'covariance'),
-    ('decision_forest_classification_batch', None, lambda r: r[1].prediction,
-     (2019, 'P', 1)),
-    ('decision_forest_regression_batch', 'decision_forest_regression_batch.csv',
-     lambda r: r[1].prediction, (2019, 1)),
+    ('decision_forest_classification_default_dense_batch',
+     None, lambda r: r[1].prediction, (2019, 'P', 1)),
```
Comment (PetrovKP) [inline tests/test_examples.py:181]: You can remove old checks (less than 2020.3)

**Review by PetrovKP — COMMENTED** [review]
don't remember to add [here](https://github.com/oneapi-src/oneDAL/blob/master/docs/source/daal/algorithms/decision_forest/decision-forest-classification.rst#examples-1)

File: tests/test_examples.py (hunk 3 above)
Comment (agorshk) [reply tests/test_examples.py]: Thanks, old versions were removed, information about hist method was added with [appropriate PR](https://github.com/oneapi-src/oneDAL/pull/1217)

**Review by PetrovKP — APPROVED** [review]

File: daal4py/sklearn/ensemble/_forest.py [hunk 4]
```diff
@@ -700,7 +723,9 @@ def __init__(self,
                  verbose=0,
                  warm_start=False,
                  ccp_alpha=0.0,
-                 max_samples=None):
+                 max_samples=None,
+                 maxBins=256,
+                 minBinSize=1):
```
Comment (michael-smirnov) [inline daal4py/sklearn/ensemble/_forest.py:728]: How this value was chosen? I see that at least in oneAPI interfaces default `minBinSize = 5`
https://github.com/oneapi-src/oneDAL/blob/master/cpp/oneapi/dal/algo/decision_forest/common.cpp#L60

File: examples/decision_forest_classification_hist_batch.py [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    print(
+        "\nDecision forest prediction results (first 10 rows):\n",
+        predict_result.prediction[0:10]
+    )
+    print(
+        "\nDecision forest probabilities results (first 10 rows):\n",
+        predict_result.probabilities[0:10]
+    )
+    print("\nGround truth (first 10 rows):\n", plabels[0:10])
+    print('All looks good!')
```
Comment (michael-smirnov) [inline examples/decision_forest_classification_hist_batch.py:92]: Can you please check that decision forest examples in `sycl` subfolder match your changes in terms of example name, parameter values, printed results, etc.?

**Review by michael-smirnov — CHANGES_REQUESTED** [review]

File: examples/decision_forest_classification_hist_batch.py (hunk 5 above)
Comment (agorshk) [reply examples/decision_forest_classification_hist_batch.py:92]: Two examples for hist method (the same method on cpu and gpu) with similar parameters were added. Old sycl examples (defaultDense for cpu and hist for gpu) were saved due to there are the only examples (for sycl) of Random Forest for version less than 2021.2, it'll be removed in the future.

File: daal4py/sklearn/ensemble/_forest.py (hunk 4 above)
Comment (agorshk) [reply daal4py/sklearn/ensemble/_forest.py:728]: It's for better accuracy for small datasets

**Review by Pahandrovich — APPROVED** [review]

**Review by michael-smirnov — APPROVED** [review]

**Pull request merged**